### DISCUSSION TITLE
File linking issue with VS Code

### DISCUSSION THREAD

**Discussion by geowa4** [BurntSushi/ripgrep#1749] [Q&A] [answered]
Building off https://github.com/BurntSushi/ripgrep/discussions/1593, I'm having trouble in VS Code with using cmd+click to open a file to a specific line when using `--no-heading` or `--vimgrep`. As an example, when I run `rg FROM`, the output includes `Dockerfile:1:1:FROM node:14`. Hovering over the file name links all the way through `FROM`, which doesn't work. Is there a way to add a space or something after `Dockerfile:1:1`? Maybe I could change the `:` via an option I couldn't find?

Alternatively, I've noticed if the file starts with a folder name, including simply `./`, VS Code handles this fine (`echo './Dockerfile:1:1:FROM node:14'` to test). Is there an option to always lead with a `./` especially if the files are in the current directory?

I'll be looking to make an issue or discussion on microsoft/vscode to address their parser greediness, too. Here's my ripgreprc for the VS Code integrated terminal in case it helps.

```
--vimgrep
--smart-case
--max-columns=150
--max-columns-preview
--glob=!git/*
--follow
--hidden

--color
always

--type-add
web:*.{html,css,js,vue}*
```

Thanks in advance,
George

**Comment by BurntSushi** [comment, marked as answer]
> Hovering over the file name links all the way through `FROM`, which doesn't work. Is there a way to add a space or something after `Dockerfile:1:1`? Maybe I could change the `:` via an option I couldn't find?

No, sorry. This has been asked many times on the issue tracker. AFAIK, some programs handle this just fine, but some don't. The ones that don't should fix their cmd+click implementation.



> Is there an option to always lead with a `./` especially if the files are in the current directory?

Yes. Use `rg foo ./` instead of `rg foo`. :-)

**Reply by geowa4** [reply]
>Use rg foo ./ instead of rg foo. :-)

Of course... 😂  Thank you for the quick reply! And yeah, I agree VS Code should fix that.
